roles/dns-records/tasks/main.yml

```diff
@@ -54,6 +54,7 @@
     nsupdate_server_public: "{{ external_nsupdate_keys['public']['server'] }}"
     nsupdate_key_secret_public: "{{ external_nsupdate_keys['public']['key_secret'] }}"
     nsupdate_key_algorithm_public: "{{ external_nsupdate_keys['public']['key_algorithm'] }}"
+    nsupdate_public_key_name: "{{ external_nsupdate_keys['public']['key_name']|default('public-' + full_dns_domain) }}"
```

Comment:
This is a good change, and I think it could be beneficial to have it for the "private" section above as well. Do you mind making a similar change above?